test/support/utils_sycl.h

```diff
@@ -36,6 +36,20 @@
 namespace TestUtils
 {
 
+#if __cplusplus >= 201703L
+#   if __SYCL_2020_NOINIT_MACRO
+#       define __noinit sycl::no_init
+#   else
+#       define __noinit sycl::noinit
```

Comment:
Could you please explain why it was implemented using macros? Why it's not using, for example?

Reply:
Ok. Why it's not a `constexpr` variable?
I mean, why not to have `constexpr auto noinit = sycl::noinit;` insead of `#define __noinit sycl::noinit`?

Reply:
`sycl::no_init` is inline constexpr variable. So we can not use it with version of standard under 17. Also it's better to make constexpr constant inline but we can not do it with version of standard under 17. So I prefer to left it as a macro.